WordPress/gutenberg
# Remove custom button and (conditionally) show single menu on Navigation route in Browse Mode

## What?


Renders a single Navigation directly on the `/navigation` route of the Site Editor if there is only a single menu. If there are multiple menus then we render a list as per `trunk`.

Closes https://github.com/WordPress/gutenberg/issues/51561

## Why?


Reasoning was outlined by Riad in https://github.com/WordPress/gutenberg/issues/51561.

This change avoids requiring consumers of the Site Editor routes from having to programmatically determine which route to send someone to if they have a single menu vs multiple menus. 

We encapsulate the behaviour and logic inside the route components and thus a consumer can simply go to `/navigation` and know they will see the "right" thing.

## How?


Extracts the code for displaying a single menu into reusable component and hooks. Then shares this between the routes for both

- `/navigation` route
- the route for a single menu

## Testing Instructions





- Make a few menus using the Nav block.
- Open Site Editor
- Go to `Navigation` in Browse Mode.
- See multiple menus listed
- Click through to check they render as expected.
- Delete all but _one_ of the menus.
- Go back to `Navigation`.
- See the single menu rendered directly.
- Check you can duplicate, rename and delete that solitary menu.

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/444434/edb96fcc-e4e4-4bc5-9468-0d0d93058de1
